pkg/auth/providers/common/usermanager.go

```diff
@@ -278,8 +278,19 @@ func (m *userManager) EnsureClusterToken(clusterName, tokenName, description, ki
 		return "", err
 	}
 	logrus.Infof("Creating token for user %v", userName)
-	token, err = m.tokens.Create(token)
-	if err != nil && !apierrors.IsAlreadyExists(err) {
+	err = wait.ExponentialBackoff(backoff, func() (bool, error) {
+		newToken, err := m.tokens.Create(token)
+		if err != nil {
+			if apierrors.IsAlreadyExists(err) {
+				return false, nil
+			}
+			return false, err
+		}
+		token = newToken
+		return true, nil
+	})
+
+	if err != nil {
```

Comment:
Linking back to original comment: https://github.com/rancher/rancher/pull/33816#pullrequestreview-718634602

We could add a note here as well about why we added a backoff.